kernel/include/kernel_structs.h

```diff
@@ -180,6 +177,8 @@ typedef struct z_kernel _kernel_t;
 extern struct z_kernel _kernel;
 
 #ifdef CONFIG_SMP
+static inline struct _cpu *z_arch_curr_cpu(void);
```

Comment:
this is also something that is defined in arch_interface.h

Reply:
I haven't fully analysed it, but the point of it is to provide a forward reference to the function, as there seemed to be some files that include kernel_structs.h first, then have some code that uses z_arch_curr_cpu, and then include arch_interface.h (yes, it is a big mess).

Reply:
I can't get behind a patch which un-does the consolidation I recently spent quite some time doing for our arch headers.

> as there seemed to be some files that include kernel_structs.h first, then have some code that uses z_arch_curr_cpu, and then include arch_interface.h 

That's what needs to be fixed.
Is this code in arch/ space or within the kernel?

Reply:
I don't remember exactly which one it was because it came up while running hundreds of sanity tests.

I will check if I can pin-point this tomorrow.